From 9c75237e18d31c436a6d0ddcf3ebd50a23cd7775 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Wed, 22 Jan 2020 08:07:02 +0100
Subject: [PATCH] nitrogen8mm-dwe: Enable necessary configs for BalenaOS

Enable necessary configs for loading and updating
resinOS_uEnv.txt

Also, increase default env size to accomodate
for inclusion of balenaOS u-boot scripts.

Add "cmd_custom" command that is lacking in current
github revision d18a2c25e47c0d71e45335772400386c021fb6cb

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 configs/dwe_2g_defconfig | 5 +++++
 include/configs/dwe.h    | 5 ++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/configs/dwe_2g_defconfig b/configs/dwe_2g_defconfig
index 3e1bb94504..09c50fa23d 100644
--- a/configs/dwe_2g_defconfig
+++ b/configs/dwe_2g_defconfig
@@ -100,3 +100,8 @@ CONFIG_USB_ETH_CDC=y
 CONFIG_USBNET_DEVADDR="00:19:b8:00:00:02"
 CONFIG_USBNET_HOST_ADDR="00:19:b8:00:00:01"
 CONFIG_USB_HOST_ETHER=y
+CONFIG_ENV_IS_NOWHERE=y
+CONFIG_CMD_SAVEENV=n
+CONFIG_FAT_WRITE=y
+CONFIG_CMD_EXPORTENV=y
+CONFIG_CMD_IMPORTENV=y
diff --git a/include/configs/dwe.h b/include/configs/dwe.h
index f5bed34abf..f438c416ad 100644
--- a/include/configs/dwe.h
+++ b/include/configs/dwe.h
@@ -59,8 +59,6 @@
 /* Flat Device Tree Definitions */
 #define CONFIG_OF_BOARD_SETUP
 
-#undef CONFIG_CMD_EXPORTENV
-#undef CONFIG_CMD_IMPORTENV
 #undef CONFIG_CMD_IMLS
 
 #undef CONFIG_CMD_CRC32
@@ -96,7 +94,7 @@
         (CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 #define CONFIG_ENV_OVERWRITE
-#define CONFIG_ENV_SIZE			0x2000
+#define CONFIG_ENV_SIZE			0x3000
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_ENV_OFFSET               (-CONFIG_ENV_SIZE)
 #define CONFIG_SYS_MMC_ENV_DEV		0	/* USDHC1 */
@@ -228,6 +226,7 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"console=" BD_CONSOLE "\0" \
+	"cmd_custom=setenv bootargs ${bootargs} mem=1G\0" \
 	"env_dev=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
 	"env_part=" __stringify(CONFIG_SYS_MMC_ENV_PART) "\0" \
 	"scriptaddr=" __stringify(CONFIG_LOADADDR) "\0" \
-- 
2.17.1

